### PR TITLE
optimize merge process

### PR DESCRIPTION
**Pull request by yhmo**
Signed-off-by: groot <yihua.mo@zilliz.com>

**What type of PR is this?**

api-change / bug / design / documentation / feature

**What this PR does / why we need it:**

**Which issue(s) this PR fixes:**

Fixes #

**Special notes for your reviewer:**

**Does this PR introduce a user-facing change?:**

**Additional documentation (e.g. design docs, usage docs, etc.):**

### CODE REVIEW

File: core/src/db/merge/MergeLayeredStrategy.cpp [hunk 1]
```diff
@@ -34,13 +43,36 @@ MergeLayeredStrategy::RegroupFiles(meta::FilesHolder& files_holder, MergeFilesGr
...
 
-    meta::SegmentsSchema& files = files_holder.HoldFiles();
+    meta::SegmentsSchema sort_files = files_holder.HoldFiles();
+    // no need to merge single file
+    if (sort_files.size() < 2) {
+        return Status::OK();
+    }
+
+    // arrange files by file size in descending order
+    std::sort(sort_files.begin(), sort_files.end(), CompareSegment);
```
Comment (JinHai-CN) [inline core/src/db/merge/MergeLayeredStrategy.cpp:53]: Why not use lambda?

File: core/src/db/merge/MergeLayeredStrategy.cpp [hunk 2]
```diff
@@ -14,11 +14,20 @@
...
 
 namespace milvus {
 namespace engine {
+namespace {
+struct {
+    bool
+    operator()(meta::SegmentSchema& left, meta::SegmentSchema& right) const {
+        return left.file_size_ > right.file_size_;
+    }
+} CompareSegment;
```
Comment (JinHai-CN) [inline core/src/db/merge/MergeLayeredStrategy.cpp:29]: This is C-style structure ...

Comment (JinHai-CN) [reply core/src/db/merge/MergeLayeredStrategy.cpp:29]: struct CompareSegment {
....

Comment (scsven) [reply core/src/db/merge/MergeLayeredStrategy.cpp:29]: "struct CompareSegment" is easier to be searched

**Review by wxyucs — CHANGES_REQUESTED** [review]

**Pull request merged**